### PR TITLE
Remove the boolean flag from `ArmeriaMessageDeframer.deframe()`

### PR DESCRIPTION
**Pull request by minwoox**
Motivation:
`ArmeriaMessageDeframer.deframe(HttpData data, boolean endOfStream)` does not have to take `endOfStream` flag
because the `HttpData` has `endOfStream` flag.

Modifications:
- Remove the boolean flag from `ArmeriaMessageDeframer.deframe()`.

Result:
- `ArmeriaMessageDeframer.deframe()` now takes `HttpData` only.

### CODE REVIEW

File: core/src/main/java/com/linecorp/armeria/common/HttpData.java [hunk 1]
```diff
@@ -61,6 +61,13 @@ static HttpData empty() {
         return ByteArrayHttpData.EMPTY;
     }
 
+    /**
+     * Returns an empty {@link HttpData} whose HTTP/2 {@code endOfStream} flag is set with the specified value.
+     */
+    static HttpData empty(boolean endOfStream) {
+        return endOfStream ? ByteArrayHttpData.EMPTY_EOS : ByteArrayHttpData.EMPTY;
+    }
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/common/HttpData.java:69]: Maybe no need to introduce this? i.e. `empty().withEndOfStream()` would work.

Comment (minwoox) [reply core/src/main/java/com/linecorp/armeria/common/HttpData.java:69]: If I didn't add this method, I thought somebody would propose this API.
Now I can completely remove this. 🤣

File: grpc/src/main/java/com/linecorp/armeria/internal/common/grpc/HttpStreamReader.java [hunk 2]
```diff
@@ -159,7 +159,7 @@ public void onNext(HttpObject obj) {
         }
         final HttpData data = (HttpData) obj;
         try {
-            deframer.deframe(data, false);
+            deframer.deframe(data);
```
Comment (ikhoon) [inline grpc/src/main/java/com/linecorp/armeria/internal/common/grpc/HttpStreamReader.java:162]: Question: Is there any chance `data` is `endOfStream`?

Comment (minwoox) [reply grpc/src/main/java/com/linecorp/armeria/internal/common/grpc/HttpStreamReader.java:162]: If the data's `endOfStream` is `true`, then we should handle it in `deframer`. 😉

**Comment by minwoox** [comment]
This could allocate an extra instance when the flag is different so let me close this. 🤣

**Pull request closed without merging**